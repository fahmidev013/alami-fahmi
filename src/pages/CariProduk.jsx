import PropTypes from "prop-types";
import React from "react";
import { useSelector, useDispatch } from "react-redux";
import { selectProductList, getProductsByKeyword } from "../redux/reducer/produkReducer";
import TableComponent from "../components/TableComponent";

export const CariProduk = () => {
  const products = useSelector(selectProductList);
  const dispatch = useDispatch();

  const onSearch = (event) => {
    event.preventDefault();
    dispatch(getProductsByKeyword(document.getElementById('cari-produk').value));
  };

  const onChange = (event) => {
    event.preventDefault();
    event.persist();
    if (event.target.value.length > 3) {
      setTimeout( () => {
        dispatch(getProductsByKeyword(event.target.value));
      }, 1500);
    }
  }

  return (
    
      <div className="site-layout-content">
          <div className="search-container">
            <input id="cari-produk" name="q" placeholder="Ketik nama Produk:" onChange={onChange} />
            <button onClick={onSearch}>Cari</button>
          </div>
          <div style={{ overflow: 'scroll', maxHeight: '65vh'}}>
            <TableComponent data={products} />
          </div>
      </div>
  );
};

CariProduk.propTypes = {
  products: PropTypes.array,
};

export default CariProduk;
