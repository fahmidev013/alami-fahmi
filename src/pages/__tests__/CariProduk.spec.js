import React from 'react';
import CariProduk from '../CariProduk';
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
const data = [
  {
    key: 25,
    id: 25,
    nama: "Alpukat",
    satuan: "kg",
    hargaSatuan: 35000,
    sellerId: 65,
    deskripsi: "Alpukat Mentega",
  },
  {
    key: 31,
    id: 31,
    nama: "Balsem",
    satuan: "botol",
    hargaSatuan: 2000,
    sellerId: 65,
    deskripsi: "Balsem Oles",
  },
];


jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn().mockReturnValue(data),
  useDispatch: jest.fn(),
}));



describe("Cari Produk Component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<CariProduk />);
  });
  it("should render with data correctly", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should call Table while data not empty ", () => {
    expect(wrapper.find("TableComponent")).toHaveLength(1);
  });

});

