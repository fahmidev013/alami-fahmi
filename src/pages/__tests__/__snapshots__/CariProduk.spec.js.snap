// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cari Produk Component should render with data correctly 1`] = `
<div
  className="site-layout-content"
>
  <div
    className="search-container"
  >
    <input
      id="cari-produk"
      name="q"
      onChange={[Function]}
      placeholder="Ketik nama Produk:"
    />
    <button
      onClick={[Function]}
    >
      Cari
    </button>
  </div>
  <div
    style={
      Object {
        "maxHeight": "65vh",
        "overflow": "scroll",
      }
    }
  >
    <TableComponent
      data={
        Array [
          Object {
            "deskripsi": "Alpukat Mentega",
            "hargaSatuan": 35000,
            "id": 25,
            "key": 25,
            "nama": "Alpukat",
            "satuan": "kg",
            "sellerId": 65,
          },
          Object {
            "deskripsi": "Balsem Oles",
            "hargaSatuan": 2000,
            "id": 31,
            "key": 31,
            "nama": "Balsem",
            "satuan": "botol",
            "sellerId": 65,
          },
        ]
      }
    />
  </div>
</div>
`;
